### ISSUE TITLE
Can `ignore-from-file` not error if file not found?

### ISSUE THREAD

**Issue by paddyroddy** [adrienverge/yamllint#611]
I'm aware this might sound like a rogue request, but I have a set of "master" pre-commit hooks where I have something like this as my `args` https://github.com/paddyroddy/.github/blob/2134df78ef3834b8af751fadc172a72ad3992d17/precommit/general/general-hooks.yml#L2-L16. However, when I use this in a repo without the `.yamllint` file, I get `FileNotFoundError: [Errno 2] No such file or directory: '.yamlignore'`. Makes perfect sense, but what if I don't mind?

**Comment by adrienverge** [comment]
If I understood correctly, I assume you meant `.yamlignore` instead of `.yamllint`, and you want yamllint not to care if this file is referenced from the config file, but doesn't exist on disk.

This use-case is interesting, but I guess most users prefer yamllint to warn them if something from their conf file is wrong or missing.

If possible, I suggest dynamically adding `ignore-from-file: .yamlignore` in your `--config-data` depending whether the file exists or not.

**Comment by paddyroddy** [comment]
> If I understood correctly, I assume you meant `.yamlignore` instead of `.yamllint`, and you want yamllint not to care if this file is referenced from the config file, but doesn't exist on disk.

Yep, sorry that was a typo, I do mean `.yamlignore`.

> This use-case is interesting, but I guess most users prefer yamllint to warn them if something from their conf file is wrong or missing.

 Completely understand why most users wouldn't want this, but I think of it akin to a missing `.gitignore` - generally advisable, but required.

> If possible, I suggest dynamically adding `ignore-from-file: .yamlignore` in your `--config-data` depending whether the file exists or not.

Hmm... Good idea. Although not sure how I'd do it in practice.

**Comment by perlpunk** [comment]
I also think that in case of `ignore-from-file` it should not complain about a missing file.
One use case is: We would like to symlink a .yamllint file from an external, common repo (either used with git-submodule or git-subrepo) that has this:
```
ignore-from-file: [.gitignore, .yamlignore]
```
and the only thing which would differ for individual repos would be the files we want to ignore. By default linking .yamllint would just work, even if .yamlignore doesn't exist.
But I can understand the reasons for the current behaviour as well.

**Comment by adrienverge** [comment]
That's interesting, thanks for the feedback :+1:

I'm curious about what more users think about this.

**Comment by 42atomys** [comment]
We currently facing the same "issue" when using a dynamic lint across multiples repositories how don't have `.gitignore` or `.yamlignore` file. 

The only workaround actually are to dynamically provide a config-data with a check of file exist before as @adrienverge say before. For us ignore this error are the good practice, but we can think about a warn log when the file are not found to alert the user (_or create a flag to toggle between error and no-error._ Not a big fan for me but that is a personal opinion)

**Comment by polyzen** [comment]
Maybe even go one step further and automatically assume the contents of .gitignore shouldn't be linted.

**Comment by polyzen** [comment]
It would be nice to be able to use `ignore-from-file` in one's global config, at least.

**Comment by RogueScholar** [comment]
I'm also interested in seeing yamllint silently ignore missing members of the `ignore-from-file` key. I'd even go so far as to say that the most robust implementation of this feature that I can imagine is to have the default behavior for missing members as silently ignore, and to allow the grammar from the `rules` map to be used with it to generate a warning or failure, at the user's discretion. To wit, the configuration file would be amended in such instances with…

## Configuration file examples

- **Display warning for missing members**

  ```yaml
  ignore-from-file: [ .gitignore, .yamlignore ]
    level: warning
  ```

- **Fail on certain missing members (exit code ≥ 1), warn on others, silently ignoring the remainder**

  ```yaml
  ignore-from-file:
    - .gitignore
    - .yamlignore
      level: error
    - "${XDG_CONFIG_HOME}/yamllint-global-ignore"
      level: warning
  ```

If the behavior directive could be scoped to either the key as a whole or each individual member, as shown above, this would conceivably cover all possible preferred outcomes. Best of all, this is accomplished without introducing any new concepts, merely extending those that already exist.
